doc.html

```diff
@@ -0,0 +1,118 @@
+- GET /feedingreport: return all feedingreports in the given time range
+
+<h2>6 Data Formats</h2>
+
+The following data model describes the 3 main entities and how they interact with the service animal_data.
+
+<img src="doc/atlas feedingmodel datamodel.png" style="width:450px" />
+
+<h3>Ration</h3>
+A ration is planned for a certain date and calculated for a single cow per day. This comes mostly out of ration calculation software and is used as starting point to programm a mixer wagon. There might be other entities like a pen included to calculate the real amount per group and per feeding.
+
+<h3>Feedstuff</h3>
+A feedstuff is a single component for a ration like grassilage or corn.
+
+<h3>FeedingReport</h3>
+This is the record from an actual feeding. It includes several entities:
+- loadings: which amount of feedstuffs are really loaded and mixed compared to the planned ration
+- unloadings: each ration might be planned for multiple pens. the unloadings describe how much of the mix was unloaded at which pen. also here compare plan and actual amounts.
+
+<h2>7 Access and Authentication</h2>
+
+Farmers MUST have an account setup at the service provider in order to authenticate and access API endpoints. The service implementor is responsible for the creation of accounts; it is not covered in the ATLAS Service Template specifications.
+
+Unless specifically documented in the OpenAPI specifications, all operation must include credentials in form of Bearer authentication (also called token authentication). Clients can obtain an access token on behalf of their user from the service's authorization server (see ATLAS service pairing).
+
+Some of the information held by animal_data services are considered to be sensitive from a GDPR perspective.The service's authorization server MUST request the client's end-user consent at service pairing time in order to deliver an access token.
+
+<h2>8 Dynamic Behaviour</h2>
+
+When a ration is send to the service there might be feedingreports when this ration is actually used some days later available at the feedingreports endpoint.
+
+<h3>Calculations</h3>
+
+The amount values in the ration are calculated to actual loading and unloading amounts in the following way:
+
+```
+// 1. Calculate freshmass sum of total loading by sum up the Pens:
+total_loading_amount = sum(pens[].feedIntakePerAnimalKg * pens[].numberOfAnimals)
+
+// 2. Calculate freshmass sum of ration advice
+total_ration_amount = sum(rationFeedstuffs[].amountAsFed)
+
+// 3. Scale up each single feedstuff with the total loading-amount
+loading.ingredients[].calculatedAmountKg = rationFeedstuffs[].amountAsFed / total_ration_amount * total_loading_amount
+
+// 4. Unloading-amounts are just used from the pens list
+unloadings[].calculatedAmountKg = pens[].feedIntakePerAnimal * pens[].numberOfAnimals
+
+```
+
+<h3>Validation</h3>
+<h4>Pen Usage</h4>
+The service has to validate, that a pen is not assigned to two rations in parallel in one query.
```

Comment:
"...not assigned to two rations..." => "...not assigned to more than one ration..."